src/Database/IdentifierQuoter.php

```diff
@@ -101,19 +104,16 @@ public function quoteExpression(ExpressionInterface $expression): void
     }
 
     /**
-     * Quotes all identifiers in each of the clauses of a query
+     * Quotes all identifiers in each of the clauses/parts of a query
      *
      * @param \Cake\Database\Query $query The query to quote.
+     * @param array $parts Query clauses.
      * @return void
      */
-    protected function _quoteParts(Query $query): void
+    protected function _quoteParts(Query $query, array $parts): void
```

Comment:
Having the parts list separate from the Query instance doesn't feel right. When would we describe a query different than itself?

Reply:
These are just the part/clause names which need to be processed. They are different depending on the query type. The actual values for these parts are fetched from the Query itself.